File: document.py

# Encapsulates a document to be printed to the laser printer.

class Document(object):
    def __init__(self, title):
        self.title = title
        self.cuts = []
        self.rasters = []

    # Object of the Cut class.
    def addCut(self, cut):
        self.cuts.append(cut)

    # Object of the Raster class.
    def addRaster(self, raster):
        self.rasters.append(raster)

    # Pixels per inch.
    def getResolution(self):
        # Max this out because we can only generate integer cut points.
        # return 1200

        # Match Fusion default.
        return 600

    # Whether this doc has engravings (raster).
    def getEnableEngraving(self):
        return len(self.rasters) > 0

    # Whether this doc has cuts (vector).
    def getEnableCut(self):
        return len(self.cuts) > 0

    def getCenterEngrave(self):
        # We never center anything automatically.
        return False

    def getAirAssist(self):
        # I assume we always want this on. If this is the air pipe aimed
        # right at the cut, then perhaps we could turn this off to reduce
        # charring.
        return True

    # Bed size.
    def getWidth(self):
        return 24*self.getResolution()

    # Bed size.
    def getHeight(self):
        return 18*self.getResolution()

    def getAutoFocus(self):
        # We never want this.
        return False

    def getTitle(self):
        return self.title

    def getCuts(self):
        return self.cuts

    def getRasters(self):
        return self.rasters

File: web/Document.js
// Encapsulates a document to be printed to the laser printer.

'use strict';

define([], function () {
    var Document = function (title) {
        this.title = title;
        this.cuts = [];
        this.rasters = [];
    };

    // Add a Cut object.
    Document.prototype.addCut = function (cut) {
        this.cuts.push(cut);
    };

    // Add a Raster object.
    Document.prototype.addRaster = function (raster) {
        this.rasters.push(raster);
    };

    // Whether this doc has engravings (raster).
    Document.prototype.getEnableEngraving = function () {
        return this.rasters.length > 0;
    };

    // Whether this doc has cuts (vector).
    Document.prototype.getEnableCut = function () {
        return this.cuts.length > 0;
    };

    Document.prototype.getCenterEngrave = function () {
        // We never center anything automatically.
        return false;
    };

    Document.prototype.getAirAssist = function () {
        // I assume we always want this on. If this is the air pipe aimed
        // right at the cut, then perhaps we could turn this off to reduce
        // charring.
        return true;
    };

    // Bed size in inches.
    Document.prototype.getWidth = function () {
        return 32;
    };

    // Bed size in inches.
    Document.prototype.getHeight = function () {
        return 20;
    };

    Document.prototype.getAutoFocus = function () {
        // We never want this.
        return false;
    };

    Document.prototype.getTitle = function () {
        return this.title;
    };

    Document.prototype.getCuts = function () {
        return this.cuts;
    };

    Document.prototype.getRasters = function () {
        return this.rasters;
    };

    Document.prototype.eachCut = function (callback) {
        _.each(this.cuts, callback);
    };

    Document.prototype.eachRaster = function (callback) {
        _.each(this.rasters, callback);
    };

    Document.prototype.addPaths = function (paths) {
        _.each(paths, function (path) {
            this.addPath(path);
        }.bind(this));
    };

    /*
    Document.prototype.addPath = function (path) {
        cut = Cut(4, 100, 50)
        # Convert to doc's resolution.
        cut.points = [Vector2(p.x*dpi/DPI, p.y*dpi/DPI) for p in path]
        doc.addCut(cut)
    };
    */

    return Document;
});
